src/System.Runtime.Analyzers/UnitTests/InitializeValueTypeStaticFieldsInlineTests.cs

```diff
@@ -12,12 +12,328 @@ public class InitializeValueTypeStaticFieldsInlineTests : DiagnosticAnalyzerTest
+Public Class Class1
+	Private Shared ReadOnly field As Integer = 1
+	Shared Sub New() ' Empty
+	End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public void VisualBasic_CA2207_NoDiagnostic_EmptyStaticConstructor()
+        {
+            VerifyBasic(@"
+Public Structure Struct1
+	Private Shared ReadOnly field As Integer = 1
+	Shared Sub New() ' Empty
+	End Sub
+End Structure
+");
+        }
+
+        [Fact]
+        public void VisualBasic_CA1810_NoDiagnostic_NoStaticFieldInitializedInStaticConstructor()
+        {
+            VerifyBasic(@"
+Public Class Class1
+	Private Shared ReadOnly field As Integer = 1
+	Shared Sub New() ' No static field initalization
+		Class1_Method()
+		Dim field2 = 1
+	End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public void Basic_CA1810_NoDiagnostic_StaticPropertyInStaticConstructor()
+        {
+            VerifyBasic(@"
+Public Class Class1
+	Private Shared ReadOnly Property [Property]() As Integer
+		Get
+			Return 0
+		End Get
+	End Property
+
+	Shared Sub New()
+		' Static property initalization
+		[Property] = 1
+	End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public void Basic_CA1810_NoDiagnostic_InitializionInNonStaticConstructor()
+        {
+            VerifyBasic (@"
+Public Class Class1
+	Private Shared ReadOnly field As Integer = 1
+	Public Sub New() ' Non static constructor
+		field = 0
+	End Sub
+
+	Public Shared Sub Class1_Method() ' Non constructor
+		field = 0
+	End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public void Basic_CA1810_Diagnostic_InitializationInStaticConstructor()
+        {
+            VerifyBasic(@"
+Public Class Class1
+	Private Shared ReadOnly field As Integer
+	Shared Sub New()
+		' Non static constructor
+		field = 0
+	End Sub
+End Class
+",
+    GetCA1810BasicDefaultResultAt(4, 13, "Class1"));
+
+        }
+
+        [Fact]
+        public void Basic_CA2207_Diagnostic_InitializationInStaticConstructor()
+        {
+            VerifyBasic(@"
+Public Structure Struct1
+	Private Shared ReadOnly field As Integer
+	Shared Sub New()
+		' Non static constructor
+		field = 0
+	End Sub
+End Structure
+",
+    GetCA2207BasicDefaultResultAt(4, 13, "Struct1"));
+
+        }
+
+        [Fact]
+        public void Basic_CA1810_Diagnostic_NoDuplicteDiagnostics()
+        {
+            VerifyBasic(@"
+Public Class Class1
+	Private Shared ReadOnly field As Integer, field2 As Integer
+	Shared Sub New()
+		' Non static constructor
+		field = 0
+		field2 = 0
+	End Sub
+End Class",
+    GetCA1810BasicDefaultResultAt(4, 13, "Class1"));
+
+        }
+
+        [Fact]
+        public void Basic_CA2207_Diagnostic_NoDuplicteDiagnostics()
+        {
+            VerifyBasic(@"
+Public Structure Struct1
+	Private Shared ReadOnly field As Integer, field2 As Integer
+	Shared Sub New()
+		' Non static constructor
+		field = 0
+		field2 = 0
+	End Sub
+End Structure",
+    GetCA2207BasicDefaultResultAt(4, 13, "Struct1"));
+
+        }
```

Comment:
No, I generally use the FxCop tests for reference, but didn't do so this time :)